pkg/skaffold/runner/build_deploy.go

```diff
@@ -39,8 +39,8 @@ func (r *SkaffoldRunner) BuildAndTest(ctx context.Context, out io.Writer, artifa
 		return nil, err
 	}
 
-	// In dry-run mode, we don't build anything, just return the tag for each artifact.
-	if r.runCtx.Opts.DryRun {
+	// In dry-run mode or when the skip build flag set to true, we don't build anything, just return the tag for each artifact.
+	if r.runCtx.Opts.DryRun || r.runCtx.Opts.SkipBuild {
```

Comment:
The major assumption here is that the tag generation is going to point to the desired tag, based on env vars, git commit, etc. And if it is pointing to `latest`, then we are assuming that the right "latest" version is pulled down. I'm not sure I'm entirely comfortable with this "magical alignment", can lead to easily misdeployed, hard to debug version mismatches. I'd rather rely on passing in the artifact tags explicitly. What do you think?

Reply:
Example: 

1. user builds image with tag `image:FOO@sha256-balint` based on an `envTemplate` tagger with `$BAR`, running skaffold with `env BAR=FOO`. 
2. user renders in a different script, different time, different CI pipeline with a mistakenly different `env BAR=BAR` --- skaffold render will create manifests with `image:BAR@sha256-somethingelse`

Or a more subtle one: 

1. user builds image with tag `image:latest@sha256-balint` based on an `sha256`
2. someone else also builds the same image _mutating_ the latest tag to `image:latest@sha256-chris`
3. user renders in a different script, different time, different CI pipeline with a mistakenly --- skaffold render will create manifests with `image:latest@sha256-chris` instead of the original intention!!! 

I believe that we should force the user to pass in the desired images to be replaced explicitly, using the digests of the images, i.e. using the output of `skaffold build`, instead of trying to infer it.

Reply:
Made a few changes base on the discussion in https://docs.google.com/document/d/1CE3PElord0pu6ycTgloL0PYAg3cXCtpfTq4Gu84wDQU/edit